forte/data/extractor/attribute_extractor.py

```diff
@@ -0,0 +1,167 @@
+
+        Returns:
+            Any. The attribute extracted from entry.
+        """
+        return getattr(entry, attr)
+
+    def set_attribute(self, entry: Entry, attr: str, value: Any):
+        r"""Set the attribute of an entry to value.
+        You can overwrite this function if you have special way to
+        set the attribute.
+
+        Args:
+            entry (Entry): An instance of Entry type, where the
+                attribute will be set.
+            attr (str): The name of the attribute.
+            value (Any): The value to be set for the attribute.
+        """
+        if attr == "text":
+            raise AttributeError("text attribute of entry cannot "
+                                "be changed.")
+        setattr(entry, attr, value)
+
+    def update_vocab(self, pack: DataPack, instance: Annotation):
+        r"""Get all attributes of one instance and
+        add them into the vocabulary.
+
+        Args:
+            pack (DataPack): The datapack that contains the current
+                instance.
+            instance (Annotation): The instance from which the
+                extractor will extractor feature.
+        """
+        for entry in pack.get(self.config.entry_type, instance):
+            element = self.get_attribute(entry, self.config.attribute)
+            if not isinstance(element, Hashable):
+                raise AttributeError("Only hashable element can be"
+                    "added into the vocabulary. Consider setting"
+                    "vocab_method to be raw and do not call update_vocab"
+                    "if you only need the raw attribute value without"
+                    "coverting them into index.")
+            self.add(element)
+
+    def extract(self, pack: DataPack, instance: Annotation) -> Feature:
+        r"""Extract attributes of one instance.
+        For example, the text of tokens in one sentence.
+
+        Args:
+            pack (Datapack): The datapack that contains the current
+                instance.
+            instance (Annotation): The instance from which the
+                extractor will extractor feature.
+
+        Returns:
+            Feature: a feature that contains the extracted data.
+        """
+        data = []
+        for entry in pack.get(self.config.entry_type, instance):
+            value = self.get_attribute(entry, self.config.attribute)
+            rep = self.element2repr(value) if self.vocab else value
+            data.append(rep)
+
+        meta_data = {"need_pad": self.config.need_pad,
+                     "pad_value": self.get_pad_value(),
+                     "dim": 1,
+                     "dtype": int if self.vocab else Any}
+
+        return Feature(data=data,
+                       metadata=meta_data,
+                       vocab=self.vocab)
+
+    def pre_evaluation_action(self, pack: DataPack, instance: Annotation):
+        r"""Remove attributes of one instance. For
```

Comment:
"Remove attributes" is not the only possible action.

You can simply say:

"This function is performed on the pack before the evaluation stage, allowing one to perform some actions to before the evaluation. By default, this function will remove the attribute and you can overwrite this ......"

Reply:
Got it.